legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/depot/api/BaseDepotApi.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.finos.legend.sdlc.server.depot.DepotServerInfo;
+import org.finos.legend.sdlc.server.depot.auth.AuthClientInjector;
+import org.finos.legend.sdlc.server.error.LegendSDLCServerException;
+import org.finos.legend.sdlc.server.tools.StringTools;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+abstract class BaseDepotApi
+{
+    private static final Logger logger = LoggerFactory.getLogger(BaseDepotApi.class);
+
+    private final DepotServerInfo serverInfo;
+    private final AuthClientInjector authClientInjector;
+
+    protected BaseDepotApi(DepotServerInfo serverInfo, AuthClientInjector authClientInjector)
+    {
+        LegendSDLCServerException.validateNonNull(authClientInjector, "Auth client injector may be null");
+
+        this.serverInfo = serverInfo;
+        this.authClientInjector = authClientInjector;
+    }
+
+    public DepotServerInfo getServerInfo()
+    {
+        return this.serverInfo;
+    }
+
+    protected URI buildURI(String path, List<NameValuePair> parameters)
+    {
+        URIBuilder builder = this.serverInfo.newURIBuilder().setPath(path);
+
+        if (parameters != null)
+        {
+            builder.addParameters(parameters);
+        }
+
+        try
+        {
+            return builder.build();
+        }
+        catch (URISyntaxException ex)
+        {
+            throw new RuntimeException("Error building depot URI: " + builder.toString(), ex);
+        }
+    }
+
+    protected String execute(HttpUriRequest request)
+    {
+        CloseableHttpClient client = this.authClientInjector.inject(HttpClientBuilder.create()).build();
+
+        try (CloseableHttpResponse response = client.execute(request))
+        {
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            switch (statusCode)
+            {
+                case HttpStatus.SC_OK:
+                {
+                    return response.getEntity() == null ? "" : EntityUtils.toString(response.getEntity(), "UTF-8");
+                }
+                case HttpStatus.SC_UNAUTHORIZED:
+                case HttpStatus.SC_FORBIDDEN:
+                {
+                    throw new DepotServerException("Authentication failed. Server responded with code " + statusCode);
+                }
+                default:
+                {
+                    throw new DepotServerException("Server responded with code " + statusCode + " when 200 was expected");
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.warn(ex.getMessage());
```

Comment:
Instead of warning with the exception message, warn with the exception. That way the stack trace will show up in the logs, which you will be glad of when trying to debug issues.

Reply:
added